src/bci_build/package/__init__.py

```diff
@@ -586,6 +586,12 @@ def __post_init__(self) -> None:
         ):
             self.support_level = SupportLevel.TECHPREVIEW
 
+    @abc.abstractmethod
+    def prepare_generation(self) -> None:
```

Comment:
nit: prepare generation of what? this name feels incomplete, but I'm not sure how to improve also. Perhaps `prepare_template_generation` or similar.

Reply:
renamed to prepare_template